src/vivarium_gates_child_iv_iron/data/utilities.py

```diff
@@ -0,0 +1,341 @@
+    # If no age group ids are specified, use the standard GBD 2019 age bins
+    if not age_group_ids:
+        age_group_ids = gbd.get_age_group_id()
+    # from gbd.get_age_bins()
+    q = f"""
+                SELECT age_group_id,
+                       age_group_years_start,
+                       age_group_years_end,
+                       age_group_name
+                FROM age_group
+                WHERE age_group_id IN ({','.join([str(a) for a in age_group_ids])})
+                """
+    raw_age_bins = query(q, 'shared')
+
+    # from utility_data.get_age_bins()
+    age_bins = (
+        raw_age_bins[['age_group_id', 'age_group_name', 'age_group_years_start', 'age_group_years_end']]
+        .rename(columns={'age_group_years_start': 'age_start', 'age_group_years_end': 'age_end'})
+    )
+
+    # set age start for birth prevalence age bin to -1 to avoid validation issues
+    age_bins.loc[age_bins['age_end'] == 0.0, 'age_start'] = -1.0
+    return age_bins
+
+
+def filter_relative_risk_to_cause_restrictions(data: pd.DataFrame) -> pd.DataFrame:
+    """ It applies age restrictions according to affected causes
+    and affected measures. If affected measure is incidence_rate,
+    it applies the yld_age_restrictions. If affected measure is
+    excess_mortality_rate, it applies the yll_age_restrictions to filter
+    the relative_risk data"""
+
+    temp = []
+    affected_entities = set(data.affected_entity)
+    affected_measures = set(data.affected_measure)
+    for cause, measure in product(affected_entities, affected_measures):
+        df = data[(data.affected_entity == cause) & (data.affected_measure == measure)]
+        cause = get_entity(EntityKey(f'cause.{cause}.{measure}'))
+        if measure == 'excess_mortality_rate':
+            start, end = vi_utils.get_age_group_ids_by_restriction(cause, 'yll')
+        else:  # incidence_rate
+            start, end = vi_utils.get_age_group_ids_by_restriction(cause, 'yld')
+        temp.append(df[df.age_group_id.isin(range(start, end + 1))])
+    data = pd.concat(temp)
+    return data
+
+
+def get_intervals_from_categories(lbwsg_type: str, categories: Dict[str, str]) -> pd.Series:
+    if lbwsg_type == "low_birth_weight":
+        category_endpoints = pd.Series(
+            {cat: parse_low_birth_weight_description(description) for cat, description in categories.items()},
+            name=f'{lbwsg_type}.endpoints'
+        )
+    elif lbwsg_type == "short_gestation":
+        category_endpoints = pd.Series(
+            {cat: parse_short_gestation_description(description) for cat, description in categories.items()},
+            name=f'{lbwsg_type}.endpoints'
+        )
+    else:
+        raise ValueError(f'Unrecognized risk type {lbwsg_type}.  Risk type must be low_birth_weight or short_gestation')
+    category_endpoints.index.name = 'parameter'
+
+    return category_endpoints
+
+
+def parse_low_birth_weight_description(description: str) -> pd.Interval:
```

Comment:
Separated out the two string parsing functions (each for low birth weight and short gestation) to get category intervals which was originally in LBWSG component.